keel-retrofit/src/main/kotlin/com/netflix/spinnaker/keel/retrofit/AuthorizationHeadersInterceptor.kt

```diff
@@ -0,0 +1,39 @@
+package com.netflix.spinnaker.keel.retrofit
+
+import com.netflix.spinnaker.kork.common.Header
+import okhttp3.Interceptor
+import okhttp3.Interceptor.Chain
+import okhttp3.Response
+import org.slf4j.LoggerFactory
+
+/**
+ * Okhttp3 interceptor that adds the X-SPINNAKER-USER and X-SPINNAKER-ACCOUNTS headers to enable authorization
+ * with downstream Spinnaker services.
+ */
+class AuthorizationHeadersInterceptor : Interceptor {
+  private val log by lazy { LoggerFactory.getLogger(javaClass) }
+
+  override fun intercept(chain: Chain): Response {
+    var request = chain.request()
+    val headers = SpinnakerCallContexts.get()[chain.call().request()]
+
+    if (headers != null) {
+      headers[Header.USER.header]?.also { xSpinnakerUser ->
```

Comment:
The interceptor is pretty straightforward: it takes the headers added to the shared call contexts map by the call adapter, and includes them in the request.